parser: do not abort instantly on '%z'/'%Z' on Windows

fujimotos opened the pull request:
Right now, the follwing command results in an error on startup on Windows:

    PS> .\bin\fluent-bit -R ..\conf\parsers.conf ...

This is just because flb_parser is too strict on tm_gmtoff support,
and would abort if SOME of the parser definitions include '%Z'/'%z'
in their Time_Format.

In truth, it's perfectly fine to use these parsers on Windows, since
our strptime implementation can handle %Z/%z without any problems. So
let's just relax the restriction here.

Part of #960

edsiper commented:
I think the correct fix will be to make sure our test works fine if there is some kind of gmtoff support (by system or by compat):

https://github.com/fluent/fluent-bit/blob/master/CMakeLists.txt#L386

fujimotos commented:
Thanks. I'll rethink this patch next week.

edsiper commented:
ping

fujimotos commented:
Sorry! I'll rework this patch this week and re-push it as soon as it gets done.

nigels-com commented:
Sounds like something potentially compiler-version specific, perhaps some unit test coverage to check for the expected %z and %Z support?

fujimotos commented:
Ok, I've updated the patch. This issue was trickier than I first thought.

 * First of all, the purpose of FLB_HAVE_GMTOFF is to check if the
   field tm_gmtoff exists in "struct tm".
 * MSVC's libc does provide "struct tm" but doesn't provide tm_gmtoff.
   So FLB_HAVE_GMTOFF is not defined on Windows.
 * However, we can still handle a timezone format specifier ('%Z'/'%z')
   on Windows, thanks to our compat strptime implementation.

The gist is that we cannot set FLB_HAVE_GMTOFF even when we use the
compat implementation since "struct tm" still lacks tm_gmtoff (if we did,
it would broke compilation).

So I updated the patch in 05b6918 to accept '%Z' or '%z' IF a) libc
has a proper gmtoff support or b) fluent-bit use flb_strptime.c.

edsiper commented:
thanks
